tests/test_bot.py

```diff
@@ -197,94 +197,41 @@ def __init__(
 
 
 class TestBotNoReq:
-    test_flag = None
-
-    @pytest.fixture(scope="function", autouse=True)
-    def reset(self):
-        self.test_flag = None
-
     @pytest.mark.parametrize("bot_class", [Bot, ExtBot])
     def test_slot_behaviour(self, bot_class, bot, mro_slots):
         inst = bot_class(bot.token)
         for attr in inst.__slots__:
             assert getattr(inst, attr, "err") != "err", f"got extra slot '{attr}'"
         assert len(mro_slots(inst)) == len(set(mro_slots(inst))), "duplicate slot"
 
-    async def test_initialize_and_shutdown(self, bot: DictExtBot, monkeypatch):
-        async def initialize(*args, **kwargs):
-            self.test_flag = ["initialize"]
-
-        async def stop(*args, **kwargs):
-            self.test_flag.append("stop")
-
-        temp_bot = Bot(token=bot.token)
-        orig_stop = temp_bot.request.shutdown
-
-        try:
-            monkeypatch.setattr(temp_bot.request, "initialize", initialize)
-            monkeypatch.setattr(temp_bot.request, "shutdown", stop)
-            await temp_bot.initialize()
-            assert self.test_flag == ["initialize"]
-            assert temp_bot.bot == bot.bot
-
-            await temp_bot.shutdown()
-            assert self.test_flag == ["initialize", "stop"]
-        finally:
-            await orig_stop()
-
-    async def test_multiple_inits_and_shutdowns(self, bot, monkeypatch):
-        self.received = defaultdict(int)
-
-        async def initialize(*args, **kwargs):
-            self.received["init"] += 1
-
-        async def shutdown(*args, **kwargs):
-            self.received["shutdown"] += 1
-
-        monkeypatch.setattr(HTTPXRequest, "initialize", initialize)
-        monkeypatch.setattr(HTTPXRequest, "shutdown", shutdown)
-
-        test_bot = Bot(bot.token)
-        await test_bot.initialize()
-        await test_bot.initialize()
-        await test_bot.initialize()
-        await test_bot.shutdown()
-        await test_bot.shutdown()
-        await test_bot.shutdown()
-
-        # 2 instead of 1 since we have to request objects for each bot
-        assert self.received["init"] == 2
-        assert self.received["shutdown"] == 2
-
-    async def test_multiple_init_cycles(self, bot):
-        # nothing really to assert - this should just not fail
-        test_bot = Bot(bot.token)
-        async with test_bot:
-            await test_bot.get_me()
-        async with test_bot:
-            await test_bot.get_me()
-
     async def test_context_manager(self, monkeypatch, bot):
+        test_flag = []
```

Comment:
If `TestBotNoReq` would inherit from `Space`, you could make `test_flag` a classvar of `Space` and use it in both classes and also move the `reset` function there. Another reason for subclassing.

Reply:
I don't know what your reaction is supposed to tell me :D My point is: with the subclassing approach, the `test_flag` thing can be used just as it was before your PR ;)

Reply:
yes, I was just wondering if the reset thing would work if it's defined in the Base class, will have to check

Reply:
not doing this, I think the nonlocal approach is better; a reset function would make sense if all the methods are using it

Reply:
If a large number is using it, that's enoguh IMO. avoids repetition